static/js/reducers/ui.js

```diff
@@ -124,10 +135,29 @@ export const INITIAL_UI_STATE: UIState = {
   couponNotificationVisibility:        false,
   navDrawerOpen:                       false,
   learnerChipVisibility:               null,
+  dialogVisibility:                    INITIAL_DIALOG_VISIBILITY_STATE
 };
 
 export const ui = (state: UIState = INITIAL_UI_STATE, action: Action) => {
   switch (action.type) {
+
+  case SHOW_DIALOG:
+    return {
+      ...state,
+      dialogVisibility: {
+        ...state.dialogVisibility,
+        [action.payload]: true
+      },
+    };
+  case HIDE_DIALOG:
+    return {
+      ...state,
+      dialogVisibility: {
+        ...state.dialogVisibility,
+        [action.payload]: false
+      },
+    };
```

Comment:
this is nice - it would be good to refactor all of the dialogs to use this system at some point (and avoid the like dozens of `showFooDialog` actions we have now).

could you remove the blank lines above and below these two `case` statements though?